src/fs/manifest/schema.rs

```diff
@@ -140,13 +130,17 @@ impl ConfigSection {
         }
     }
 
-    async fn load_monitor(&self, args: &RunSubcommand) -> Result<BoxedMonitor> {
+    async fn load_monitor(
+        &self,
+        args: &RunSubcommand,
+        ingress: &BoxedIngress,
+    ) -> Result<BoxedMonitor> {
         // Having cloudflare configured is mutually exclusive with having
         // ingress configured. Error if both are set.
         match (&self.cloudflare, &self.monitor) {
             (Some(_), Some(_)) => Err(CloudflareMutuallyExclusiveConfig.into()),
             (None, None) => Err(MissingMonitorConfig.into()),
-            (None, Some(monitor)) => monitor.load_monitor(args).await,
+            (None, Some(monitor)) => monitor.load_monitor(args, ingress).await,
```

Comment:
Why do you need to pass in the `ingress` to load the `monitor`? That doesn't seem right.